src/ty_module/generic_arguments.md

```diff
@@ -0,0 +1,126 @@
+# ADTs and Generic Arguments
+
+## ADTs Representation
+
+Let's consider the example of a type like `MyStruct<u32>`, where `MyStruct` is defined like so:
+
+```rust,ignore
+struct MyStruct<T> { x: u8, y: T }
+```
+
+The type `MyStruct<u32>` would be an instance of `TyKind::Adt`:
+
+```rust,ignore
+Adt(&'tcx AdtDef, GenericArgs<'tcx>)
+//  ------------  ---------------
+//  (1)            (2)
+//
+// (1) represents the `MyStruct` part
+// (2) represents the `<u32>`, or "substitutions" / generic arguments
+```
+
+There are two parts:
+
+- The [`AdtDef`][adtdef] references the struct/enum/union but without the values for its type
+  parameters. In our example, this is the `MyStruct` part *without* the argument `u32`.
+  (Note that in the HIR, structs, enums and unions are represented differently, but in `ty::Ty`,
+  they are all represented using `TyKind::Adt`.)
```

Comment:
oh, for the previous comment, you kinda do that here. Maybe say upfront: ADT = struct, enum, union